pontoon/sync/tests/test_utils.py

```diff
@@ -0,0 +1,7 @@
+
+from pontoon.sync.utils import uses_undercore_as_separator
+dir = "pontoon/sync/tests/dummy_directory"
```

Comment:
We need an absolute path rather than a relative one. I suggest you define it [along with other paths](https://github.com/mozilla/pontoon/blob/408655961d7577dc5e6d7b80c3ff258e6915682c/pontoon/sync/tests/__init__.py#L22) and then import it from `pontoon.sync.tests`.

Reply:
Thanks, made a new commit where I tried making a variable with `os.path.abspath()`, but unfortunately pytest failed again. Passes on my local and when I do `print(os.path.abspath("pontoon/sync/tests/dummy_directory"))` I get the correct absolute directory for my machine. 

Also I ran `make black` and 100+ files on my local branch updated, so I specifically only added the two files where I made changes for the issue rather than having all of them on there. Could that be causing that test to still fail?

Reply:
I'm not sure abspath is the right function to use here, so to stay on the safe side, I'd do what other PATHs do, i.e.:
```python
DUMMY_DIRECTORY_PATH = os.path.join(
    os.path.dirname(__file__),
    "dummy_directory",
)
```

But more importantly - as of this patch `uses_undercore_as_separator()` only compares underscore and dash counts in directories, so you need to also add directories. Note that git cannot track empty directories, so you also need to put empty `.gitignore` files in each of them.

I'd also give `dummy_directory` a more meaningful name, e.g. `locale-separator-tests` (mind the dash instead of underscore for consistency with other folder names).

Once that's all done, you can run `make black`, but please note that we have updated the library recently, so you need to rebase your branch from upstream and run `make build-server` to install it first and only then run `make black`. Sorry for the inconvenience.

Reply:
For the empty directories, did you mean to say add an empty `.gitkeep` in each? Saw [this](https://www.theserverside.com/blog/Coffee-Talk-Java-News-Stories-and-Opinions/gitkeep-push-empty-folders-git-commit) explainer on the issue and sounds like what I need to do. I assume adding empty `.gitignore` would make the directories still be ignored.

Reply:
It doesn't matter how the file is called, this is just a workaround git's inability to track empty directories. See this explanation:
https://stackoverflow.com/a/7229996

I propose `.gitignore`, because we already use it elsewhere:
https://github.com/mozilla/pontoon/tree/master/media